packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -225,6 +225,15 @@ export class Cluster extends Resource implements ICluster {
       this.enableFargateCapacityProviders();
     }
 
+    if (this._capacityProviderNames.length > 0 && props.defaultCapacityProviderStrategy && props.defaultCapacityProviderStrategy.length > 0) {
+      const defaultCapacityProviders = props.defaultCapacityProviderStrategy.map(cp => cp.capacityProvider);
```

Comment:
I don't think we should do this map before the `forEach` call. Instead, we can call `forEach` directly on `props.defaultCapacityProviderStrategy`. 

Let's also ditch the `findIndex` in favor of `includes`

```ts
props.defaultCapacityProviderStrategy.forEach(dcp => {
  if (!this._capacityProviderNames.includes(dcp.capacityProvider) {
    throw new Error(...);
  }
}
```